src/HttpServer/HttpServer.cc

```diff
@@ -54,39 +60,41 @@ HttpServer::HttpServer(std::shared_ptr<SessionManager> session_manager, fs::path
 void HttpServer::RegisterRoutes() {
     uWS::App& app = _session_manager->App();
 
+    const std::string scripting("/api/scripting/action");
     if (_enable_scripting) {
-        app.post("/api/scripting/action", [&](auto res, auto req) { HandleScriptingAction(res, req); });
+        app.post(scripting, [&](auto res, auto req) { HandleScriptingAction(res, req); });
     } else {
-        app.post("/api/scripting/action", [&](auto res, auto req) { NotImplemented(res, req); });
+        app.post(scripting, [&](auto res, auto req) { NotImplemented(res, req); });
     }
 
+    const std::string database("/api/database/");
```

Comment:
This is unnecessary and should be reverted. The repetition problem in this block of code is not the repetition of this string (and substituting concatenation with a variable is also repetitive -- the concatenation just makes it slightly harder to read).

The problem is that identical code is repeated for the three object types, and that is what should be eliminated. Please use a foreach loop over `LAYOUT`, `SNIPPET` and `WORKSPACE` to combine the three blocks of routes for objects into one, and please use `fmt::format` for the concatenation. So for example the first line would read `app.get(fmt::format("/api/database/list/{}s", object_type), [&](auto res, auto req) { HandleGetObjectList(object_type, res, req); });` (assuming you use `object` as your loop variable, like you did below).